src/api/app/models/user.rb

```diff
@@ -674,6 +674,11 @@ def delete!
       project.destroy
     end
 
+    # comments are non-critical personal content, drop them as well. Even though they shouldn't be visible anymore
+    Comment.where(user: self).delete_all
```

Comment:
`delete_all` will skip the callbacks defined [here](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/models/comment.rb#L17) is that what you want to do?